lib/internal/crypto/random.js

```diff
@@ -124,6 +125,12 @@ function randomFill(buf, offset, size, callback) {
 // e.g.: Buffer.from("ff".repeat(6), "hex").readUIntBE(0, 6);
 const RAND_MAX = 0xFFFF_FFFF_FFFF;
 
+// Cache random data to use in randomInt.
+const randomCache = new FastBuffer(3072);
+let randomCacheOffset = randomCache.length;
+let asyncCacheFillInProgress = false;
+const asyncCachePendingTasks = [];
```

Comment:
using an array for this is usually not performing as expected for high numbers. Using a linked list solves this problem.

Reply:
we have `fixed-queue` which also might be appropriate?